### ISSUE TITLE
tokio-run-time index out of bounds on eth_getTransactionReceipt.

### ISSUE THREAD

**Issue by sunwrobert** [foundry-rs/foundry#3004]
### Component

Anvil

### Have you ensured that all of these are up to date?

- [ ] Foundry
- [ ] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (7278695 2022-08-27T00:28:29.89913Z)

### What command(s) is the bug in?

anvil

### Describe the bug

Been seeing instances of this occur sproadically in our tests. Don't have a realible way of repro-ing but wondering if the logs hint at anything.

```
api_2948024161_undefined-ethereum-mainnet-anvil-1  | thread 'tokio-runtime-worker' panicked at 'index out of bounds: the len is 1 but the index is 3', anvil/src/eth/backend/mem/mod.rs:1445:27
api_2948024161_undefined-ethereum-mainnet-anvil-1  | stack backtrace:
api_2948024161_undefined-ethereum-mainnet-anvil-1  | [2m2022-08-29T11:17:44.756569Z[0m [32m INFO[0m [2mnode::user[0m[2m:[0m eth_getTransactionReceipt
api_2948024161_undefined-ethereum-mainnet-anvil-1  |    0:     0x563a9f542d00 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |    1:     0x563a9f23420c - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |    2:     0x563a9f520740 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |    3:     0x563a9f5439d5 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |    4:     0x563a9f5447d4 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |    5:     0x563a9f544372 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |    6:     0x563a9f5442e6 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |    7:     0x563a9f5442a2 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |    8:     0x563a9eed5e62 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |    9:     0x563a9eed5ee1 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   10:     0x563a9efbee4b - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   11:     0x563a9efa18f5 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   12:     0x563a9efa027e - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   13:     0x563a9ef6f906 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   14:     0x563a9ef6d1a1 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   15:     0x563a9f1bc031 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   16:     0x563a9f1b8be8 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   17:     0x563a9ef48805 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   18:     0x563a9ef48546 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   19:     0x563a9ef48450 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   20:     0x563a9ef49000 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   21:     0x563a9ef48805 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   22:     0x563a9ef48546 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   23:     0x563a9ef48450 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   24:     0x563a9ef49000 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   25:     0x563a9ef48805 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   26:     0x563a9ef48546 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   27:     0x563a9ef4d3fa - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   28:     0x563a9ef6a3bc - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   29:     0x563a9ef48805 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   30:     0x563a9ef48546 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   31:     0x563a9ef4b086 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   32:     0x563a9ef4bbd5 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   33:     0x563a9ef48805 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   34:     0x563a9ef2cb3e - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   35:     0x563a9eef4fb3 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   36:     0x563a9ef2913e - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   37:     0x563a9ef24378 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   38:     0x563a9f55b13d - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   39:     0x563a9f55a12b - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   40:     0x563a9f288d03 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   41:     0x563a9f55f164 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   42:     0x563a9f560425 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   43:     0x563a9f545ae5 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   44:     0x7fe0b3ae5ea7 - start_thread
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   45:     0x7fe0b38cbdef - clone
api_2948024161_undefined-ethereum-mainnet-anvil-1  |   46:                0x0 - <unknown>
api_2948024161_undefined-ethereum-mainnet-anvil-1  | ./scripts/run_ethereum.sh: line 4:    10 Aborted                 (core dumped) RUST_LOG=backend,node,miner,rpc=warn RUST_BACKTRACE=full anvil --host 0.0.0.0 --fork-url="https://eth-mainnet.g.alchemy.com/v2/XXX" --fork-block-number=15347924 --port=8545 --block-time=4 --chain-id=1 --timeout=3000 --retries=5
```

**Comment by mattsse** [comment]
this looks like we're messing up the transaction index counter somewhere, checking

**Comment by mattsse** [comment]
thx for flagging, this was indeed a smol issue with how the index was computed, fixed in #3008
